include/soci/session.h

```diff
@@ -43,7 +43,7 @@ class SOCI_DECL session
 {
 private:
 
-    void set_query_transformation_(cxx_details::auto_ptr<details::query_transformation_function>& qtf);
+    void set_query_transformation_(std::unique_ptr<details::query_transformation_function>&& qtf);
```

Comment:
`unique_ptr`s should be passed by value to signify ownership transfer. And if you don't want to transfer ownership, you should pass the raw pointer instead of the smart-pointer (for semantic and performance reasons)

Reply:
Passing by value or r-value reference is the same thing for `unique_ptr` and, also, this is a private function and so we don't really care about what it takes, so I don't see any really compelling reason to change this. And of course we do want ownership transfer here.